tests/functional/tokens/test_generate.py

```diff
@@ -0,0 +1,231 @@
+def the_admin_has_the_admin_password(get_admin_pwd):
+    return get_admin_pwd
+
+
+@given(
+    "the admin gets an 'access_token' by logging in to '/api/v1/token' with a "
+    "'write:token' scope",
+    target_fixture="access_token",
+)
+def the_admin_has_generated_an_access_token_with_write_token(access_token):
+    return access_token
+
+
+@given(
+    parse("the admin adds Authorization Bearer {token} in the 'headers'"),
+    target_fixture="headers",
+)
+def the_admin_adds_authorization_token_in_the_headers(access_token, token):
+    if token == "'access_token'":
+        header_token = f"Bearer {access_token}"
+    else:
+        header_token = f"Bearer {token}"
+    headers = {"Authorization": header_token}
+    return headers
+
+
+@given(
+    parse(
+        "the admin adds JSON payload with scopes: {scopes} and "
+        "expires: {expires}"
+    ),
+    target_fixture="payload",
+)
+def the_admin_adds_payload(scopes, expires):
+    if scopes == "None":
+        scopes = None
+        # convert "None" to None
+        payload = {"scopes": scopes, "expires": expires}
+    else:
+        # convert "['str', 'str']" to ['str', 'str']
+        payload = {"scopes": ast.literal_eval(scopes), "expires": expires}
+
+    return payload
+
+
+@when(
+    "the admin sends a POST request to '/api/v1/token/new'",
+    target_fixture="response",
+)
+def the_admin_sends_request(http_request, headers, payload):
+    response = http_request(
+        method="POST", url="/api/v1/token/new", headers=headers, json=payload
+    )
+    return response
+
+
+@then("the admin should get status code '200'")
+def the_admin_gets_status_code_200(response):
+    assert response.status_code == 200, response.text
+
+
+@then("the admin should get 'access_token' with a new token")
+def the_admin_gets_access_token(response):
+    assert response.json()["access_token"] is not None
+
+
+@scenario(
+    "../../features/tokens/generate.feature",
+    "Admin cannot generate Token using HTTP API with invalid expires",
+)
+def test_admin_cannot_generate_token_using_api_with_invalid_expires():
+    """Admin cannot generate Token using HTTP API with invalid expires."""
+
+
+@when(
+    "the admin sends a POST request to '/api/v1/token/new' with invalid "
+    "'expires' in 'payload'",
+    target_fixture="response",
+)
+def the_admin_sends_with_invalid_expires_in_payload(
+    http_request, headers, payload
+):
+    response = http_request(
+        method="POST", url="/api/v1/token/new", headers=headers, json=payload
+    )
+    return response
+
+
+@then("the admin should get status code '422'")
+def the_admin_should_get_status_code_422_invalid_expires(response):
+    assert response.status_code == 422, response.text
+
+
+@then("the admin should get not allowed scope as a given context detail")
+def the_admin_should_get_not_allowed_scope_as_a_given_context_detail(response):
+    assert response.json() == {}
```

Comment:
It's a little hard to understand the connection between `the admin should get not allowed scope as a given context detail"` and the check `assert response.json() == {}`.
Can you leave a comment and describe how they are connected?

Reply:
It should not be there. I will remove it.